src/odemis/gui/util/img.py

```diff
@@ -917,29 +917,38 @@ def calculate_ticks(value_range, client_size, orientation, tick_spacing):
     return tick_list, vtp_ratio
 
 
-def draw_scale(ctx, value_range, client_size, orientation, tick_spacing,
-               fill_colour, unit, scale_width, font_size, scale_label=None, mirror=False):
+def draw_scale(value_range, client_size, orientation, tick_spacing,
+               fill_colour, unit, font_size, scale_label=None, mirror=False):
     """
     Draws horizontal or vertical scale bar
 
     value_range (tuple of floats): value range
     client_size (int, int): number of pixels in X/Y
     orientation (int): legend orientation
     tick_spacing (float): space between ticks
-    fill_colour (tuple of floats): colour to fill bars
-    unit (string): scale unit
-    scale_width (float): scale bar width
+    fill_colour (tuple of floats): colour to draw bar and tick
+    unit (str): scale unit
     font_size (float)
-    scale_label (string): label to be attached
+    scale_label (str or None): label to be attached
     mirror (boolean): if True: in case of horizontal bar means scale bar goes to the
         top side of the plot and in case of vertical scale bar goes to the right side
         of the plot. The tick values is not written.
+    return (numpy array of shape YXC with uint8): BGRA image containing the scale bar
     """
-
     if value_range is None:
         logging.info("Not drawing scale bar, as range is None")
         return
 
+    # TODO: Instead of just the min/max value, support a whole list of values,
+    # so that if they are not linearly distributed, the ticks are shown at the
+    # right place.
+
+    im = numpy.empty((client_size[1], client_size[0], 4), dtype=numpy.uint8)
+    im.fill(255)
```

Comment:
I think this can be replaced with `im = numpy.full((client_size[1], client_size[0], 4), 255, dtype=numpy.uint8)`
Or is this faster?

Reply:
Indeed, it would be clearer this way. Actually, I've been looking for such function in numpy for a long time, and never found it!